[타이타닉호 승선인원 목록 데이터 분석 및 시각화]
1. 데이터 분석 방법 : 확증적분석
2. 목적: 가설 선정과 그 검증.<br>
        가설1. 아메리칸 드림을 꿈꾸고 유럽 각지에서 모인 독신 남성이 많이 탐<br>
        가설2. 여성과 어린아이/노약자의 생존율은 높고 남성의 생존율은 낮다.<br>
3. 목표 데이터: 가설1. adult male/alone<br>
              가설2. survived - woman/age/male<br>
4. feature/attribute : 나머지 모두.



[1] 모듈 로딩 및 데이터 준비 <hr>

In [1]:
## [1-1] 모듈 로딩 
import pandas as pd                     # 데이터 분석 및 전처리용 모듈
import numpy as np                      # 배열(array)을 지원하는 파이썬 패키지

import matplotlib.pyplot as plt         # 데이터 시각화 즉, 그래프 관련 모듈 
import utils                            # 데이터확인용 

In [2]:
## [1-2] 데이터 준비
D_FILE = '../DATA/titanic.csv'
data1 = pd.read_csv(D_FILE)
ttDF = pd.DataFrame(data1)


[2] 데이터 확인<hr>

In [3]:
ttDF.head(3)

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True

In [4]:
ttDF.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   survived     891 non-null    int64  
 1   pclass       891 non-null    int64  
 2   sex          891 non-null    object 
 3   age          714 non-null    float64
 4   sibsp        891 non-null    int64  
 5   parch        891 non-null    int64  
 6   fare         891 non-null    float64
 7   embarked     889 non-null    object 
 8   class        891 non-null    object 
 9   who          891 non-null    object 
 10  adult_male   891 non-null    bool   
 11  deck         203 non-null    object 
 12  embark_town  889 non-null    object 
 13  alive        891 non-null    object 
 14  alone        891 non-null    bool   
dtypes: bool(2), float64(2), int64(4), object(7)
memory usage: 92.4+ KB


In [5]:
utils.printUniqueValue(ttDF)


[survived컬럼의 고유값]=====
갯수 :  2
[0 1]
survived
0    549
1    342
Name: count, dtype: int64

[pclass컬럼의 고유값]=====
갯수 :  3
[3 1 2]
pclass
3    491
1    216
2    184
Name: count, dtype: int64

[sex컬럼의 고유값]=====
갯수 :  2
['male' 'female']
sex
male      577
female    314
Name: count, dtype: int64

[age컬럼의 고유값]=====
갯수 :  88
[22.   38.   26.   35.     nan 54.    2.   27.   14.    4.   58.   20.
 39.   55.   31.   34.   15.   28.    8.   19.   40.   66.   42.   21.
 18.    3.    7.   49.   29.   65.   28.5   5.   11.   45.   17.   32.
 16.   25.    0.83 30.   33.   23.   24.   46.   59.   71.   37.   47.
 14.5  70.5  32.5  12.    9.   36.5  51.   55.5  40.5  44.    1.   61.
 56.   50.   36.   45.5  20.5  62.   41.   52.   63.   23.5   0.92 43.
 60.   10.   64.   13.   48.    0.75 53.   57.   80.   70.   24.5   6.
  0.67 30.5   0.42 34.5  74.  ]
age
24.00    30
22.00    27
18.00    26
19.00    25
28.00    25
         ..
36.50     1
55.50     1
0.92      1
23.50     1
74.00     1
Name: count, Le

In [6]:
# ttDF.isna().sum()
# deck 는 688/891 이 nan이므로 데이터의 가치가 없음.
ttDF = ttDF.drop(columns='deck').copy()

In [7]:
ttDF[ttDF['age'].isna()]

# 'age'에도 NaN이 많지만, 비율이 상대적으로 적으므로 보류.
# 처리하기 편하도록 nan상태를 유지


survived  pclass     sex  age  sibsp  parch     fare embarked   class  \
5           0       3    male  NaN      0      0   8.4583        Q   Third   
17          1       2    male  NaN      0      0  13.0000        S  Second   
19          1       3  female  NaN      0      0   7.2250        C   Third   
26          0       3    male  NaN      0      0   7.2250        C   Third   
28          1       3  female  NaN      0      0   7.8792        Q   Third   
..        ...     ...     ...  ...    ...    ...      ...      ...     ...   
859         0       3    male  NaN      0      0   7.2292        C   Third   
863         0       3  female  NaN      8      2  69.5500        S   Third   
868         0       3    male  NaN      0      0   9.5000        S   Third   
878         0       3    male  NaN      0      0   7.8958        S   Third   
888         0       3  female  NaN      1      2  23.4500        S   Third   

       who  adult_male  embark_town alive  alone  
5      man        True   Queenstown    no   True  
17     man        True  Southampton   yes   True  
19   woman       False    Cherbourg   yes   True  
26     man        True    Cherbourg    no   True  
28   woman       False   Queenstown   yes   True  
..     ...         ...          ...   ...    ...  
859    man        True    Cherbourg    no   True  
863  woman       False  Southampton    no  False  
868    man        True  Southampton    no   True  
878    man        True  Southampton    no   True  
888  woman       False  Southampton    no  False  

[177 rows x 14 columns]

[3] 데이터 분석 <hr>

In [8]:
tsDF = ttDF[ttDF['survived']==1]
utils.printUniqueValue(tsDF)
# print(ttDF['survived'].value_counts())


[survived컬럼의 고유값]=====
갯수 :  1
[1]
survived
1    342
Name: count, dtype: int64

[pclass컬럼의 고유값]=====
갯수 :  3
[1 3 2]
pclass
1    136
3    119
2     87
Name: count, dtype: int64

[sex컬럼의 고유값]=====
갯수 :  2
['female' 'male']
sex
female    233
male      109
Name: count, dtype: int64

[age컬럼의 고유값]=====
갯수 :  65
[38.   26.   35.   27.   14.    4.   58.   55.     nan 34.   15.   28.
  3.   19.   49.   29.   21.    5.   17.   32.    0.83 30.   33.   23.
 32.5  12.   24.   22.   16.   40.    9.    1.   45.   44.   18.   31.
  8.   37.   50.   25.   41.   63.   42.    0.92 36.    2.   60.   39.
 13.   52.   48.    0.75 54.    7.   62.   53.   20.   80.   56.    6.
  0.67 51.   43.   11.    0.42 47.  ]
age
24.00    15
36.00    11
35.00    11
27.00    11
22.00    11
         ..
55.00     1
32.50     1
12.00     1
0.92      1
47.00     1
Name: count, Length: 65, dtype: int64

[sibsp컬럼의 고유값]=====
갯수 :  5
[1 0 4 3 2]
sibsp
0    210
1    112
2     13
3      4
4      3
Name: count, dtype: int64

[parc

In [9]:
# survived/alive 기준으로 정렬
# 생존률과 상관 계수가 높은 컬럼 찾기
# 생존자 342명.
# 객실등급 1 : 136, 3: 119, 2: 87
# 성별 여성 : 233, 남성 109명
# 연령 : 다양함 - 카테고리
# 동승자/형제/배우자 0: 210, 1: 112, 2:13, 3:4, 4:3
# 동승자/부모/자녀 0: 233, 1: 65, 2:40, 3:3, 5:1
# 요금 다양함. - 카테고리로 간주해야할듯.
# 탑승항 : Southamton: 217, Cherbourg: 93, Queenstown: 30
# who : woman 205, man 88, child 49
# 성인남성 여부 : False 254, True 88
# alone True 163, False 179

In [10]:
tdDF = ttDF[ttDF['survived']==0]
utils.printUniqueValue(tdDF)


[survived컬럼의 고유값]=====
갯수 :  1
[0]
survived
0    549
Name: count, dtype: int64

[pclass컬럼의 고유값]=====
갯수 :  3
[3 1 2]
pclass
3    372
2     97
1     80
Name: count, dtype: int64

[sex컬럼의 고유값]=====
갯수 :  2
['male' 'female']
sex
male      468
female     81
Name: count, dtype: int64

[age컬럼의 고유값]=====
갯수 :  77
[22.  35.   nan 54.   2.  20.  39.  14.  31.   8.  19.  40.  66.  28.
 42.  21.  18.  27.   7.  65.  28.5 11.  45.   4.  26.  32.  16.  25.
 24.  29.  46.  59.  71.  34.  33.  37.  38.  47.  14.5 17.  70.5 32.5
 23.   9.  36.5 51.  55.5 40.5 30.  44.   1.  61.  56.  50.  36.  45.5
 20.5 62.  41.  52.  23.5 43.  15.   3.  10.  64.  48.  58.  55.  49.
 57.  70.  24.5 60.  30.5  6.  34.5 74. ]
age
21.0    19
28.0    18
25.0    17
18.0    17
22.0    16
        ..
15.0     1
36.5     1
23.5     1
55.5     1
74.0     1
Name: count, Length: 77, dtype: int64

[sibsp컬럼의 고유값]=====
갯수 :  7
[1 0 3 4 2 5 8]
sibsp
0    398
1     97
4     15
2     15
3     12
8      7
5      5
Name: count, dtype: 

In [11]:
##사망자: 549
## 객실등급 : 3:372, 2:97, 1:80
## 성별 : 남자: 468, 여자: 81
## 탑승항 : Southamton: 427, Cherbourg: 75, Queenstown: 47
## who : man: 449, woman: 66, child: 34
## 성인남성 : True: 449, False: 100
## alone: True: 374, False: 175

In [34]:
# 승객분석
# 남성 & 성인 남성 & 단독 승선
ttDF[ (ttDF['who'] == 'man') & (ttDF['adult_male'] == True)& (ttDF['alone']== True)]
# 410. 절반에 육박하는 수치. 107.

# ttDF[ (ttDF['who'] == 'man') & (ttDF['adult_male'] == True)] #& (ttDF['alone']== True)]
# # 단독을 빼면 537

# ttDF[ (ttDF['who'] == 'man')]
# 그냥 남성 537    이중 나이 미상 124

survived  pclass   sex   age  sibsp  parch     fare embarked   class  \
4           0       3  male  35.0      0      0   8.0500        S   Third   
5           0       3  male   NaN      0      0   8.4583        Q   Third   
6           0       1  male  54.0      0      0  51.8625        S   First   
12          0       3  male  20.0      0      0   8.0500        S   Third   
17          1       2  male   NaN      0      0  13.0000        S  Second   
..        ...     ...   ...   ...    ...    ...      ...      ...     ...   
883         0       2  male  28.0      0      0  10.5000        S  Second   
884         0       3  male  25.0      0      0   7.0500        S   Third   
886         0       2  male  27.0      0      0  13.0000        S  Second   
889         1       1  male  26.0      0      0  30.0000        C   First   
890         0       3  male  32.0      0      0   7.7500        Q   Third   

     who  adult_male  embark_town alive  alone  
4    man        True  Southampton    no   True  
5    man        True   Queenstown    no   True  
6    man        True  Southampton    no   True  
12   man        True  Southampton    no   True  
17   man        True  Southampton   yes   True  
..   ...         ...          ...   ...    ...  
883  man        True  Southampton    no   True  
884  man        True  Southampton    no   True  
886  man        True  Southampton    no   True  
889  man        True    Cherbourg   yes   True  
890  man        True   Queenstown    no   True  

[410 rows x 14 columns]

In [35]:
#나이대 분석
ttDF['age'].nunique()
ttDF['age'].isna().sum()

# 전체에서 나이미상 177, 이중 성인남성/단독 = 107임.

177

In [45]:
# 성인남성/단독 ama
ttDFama = ttDF[ (ttDF['who'] == 'man') & (ttDF['adult_male'] == True)& (ttDF['alone']== True)]

ttDFama['class'].value_counts()

# class
# Third     263
# First      75
# Second     72

class
Third     263
First      75
Second     72
Name: count, dtype: int64

In [47]:
ttDFama['alive'].value_counts()
# alive
# no     346
# yes     64

alive
no     346
yes     64
Name: count, dtype: int64

[3] 데이터분석<hr>


1. 아메리칸 드림을 노린 독신 남성이 많았다. = fact
실제 독신 남성 숫자는 889 데이터중 절반에 육박하는 410명이다.

2. 생존자 중 성인 남성은 적고, 여성과 어린이의 생존율은 높다.  